runtime-core/core-types/definition/mixed.inl

```diff
@@ -1800,87 +170,24 @@ T &mixed::empty_value() noexcept {
   return value;
 }
 
-namespace impl_ {
-
-template<class MathOperation>
-inline mixed do_math_op_on_vars(const mixed &lhs, const mixed &rhs, MathOperation &&math_op) {
-  if (likely(lhs.is_int() && rhs.is_int())) {
-    return math_op(lhs.as_int(), rhs.as_int());
-  }
-
-  const mixed arg1 = lhs.to_numeric();
-  const mixed arg2 = rhs.to_numeric();
-
-  if (arg1.is_int()) {
-    if (arg2.is_int()) {
-      return math_op(arg1.as_int(), arg2.as_int());
-    } else {
-      return math_op(static_cast<double>(arg1.as_int()), arg2.as_double());
-    }
-  } else {
-    if (arg2.is_int()) {
-      return math_op(arg1.as_double(), static_cast<double>(arg2.as_int()));
-    } else {
-      return math_op(arg1.as_double(), arg2.as_double());
-    }
-  }
-}
-
-} // namespace impl_
-
-inline mixed operator+(const mixed &lhs, const mixed &rhs) {
-  if (lhs.is_array() && rhs.is_array()) {
-    return lhs.as_array() + rhs.as_array();
-  }
-
-  return impl_::do_math_op_on_vars(lhs, rhs, [](const auto &arg1, const auto &arg2) { return arg1 + arg2; });
-}
-
-inline mixed operator-(const mixed &lhs, const mixed &rhs) {
-  return impl_::do_math_op_on_vars(lhs, rhs, [](const auto &arg1, const auto &arg2) { return arg1 - arg2; });
-}
-
-inline mixed operator*(const mixed &lhs, const mixed &rhs) {
-  return impl_::do_math_op_on_vars(lhs, rhs, [](const auto &arg1, const auto &arg2) { return arg1 * arg2; });
-}
-
-inline mixed operator-(const string &lhs) {
-  mixed arg1 = lhs.to_numeric();
-
-  if (arg1.is_int()) {
-    arg1.as_int() = -arg1.as_int();
-  } else {
-    arg1.as_double() = -arg1.as_double();
-  }
-  return arg1;
-}
-
-inline mixed operator+(const string &lhs) {
-  return lhs.to_numeric();
-}
-
-inline int64_t operator&(const mixed &lhs, const mixed &rhs) {
-  return lhs.to_int() & rhs.to_int();
-}
-
-inline int64_t operator|(const mixed &lhs, const mixed &rhs) {
-  return lhs.to_int() | rhs.to_int();
-}
-
-inline int64_t operator^(const mixed &lhs, const mixed &rhs) {
-  return lhs.to_int() ^ rhs.to_int();
+inline void mixed::reset_empty_values() noexcept {
```

Comment:
Why `mixed.inl` contains non template functions that can be declared in `mixed_decl.inl`? Such as `is_bool`, `is_string`, `reset_empty_values` and so on

Reply:
I would move all definitions from `mixed_decl.inl` to `mixed.inl` and check this for `string` files too, but I gonna do it in another PR